File: LaboratoryWork/Semester_2/Laboratory_7/Laboratory_2/PHP/task3.php
<?php
// Включим отображение ошибок
error_reporting(E_ALL);
ini_set("display_errors", 1);


function is_number(int $a) {
    /* Проверяет, что это число, а не цифра. */
    return ((int)($a / 10) != 0);
}


function cnt_odd_digits(int $a) {
    /* Считает количество нечетных цифр в числе. */
    $res = 0;
    $a = abs($a);

    while ($a > 0) {
        $res += ($a % 2);
        $a = (int)($a / 10);
    }
    return $res;
}


function main() {
    $n = (int)readline("Введите количество чисел: ");

    for ($i = 0; $i < $n; $i++) {
        $num = (int)readline("Введите " . ($i + 1) . " число: ");
        $cnt_odd = 0;
        
        if (is_number($num)) {
            $cnt_odd = cnt_odd_digits($num);
        } else {
            print($num . " - это цифра!");
        }

        print("Количество нечетных цифр: " . $cnt_odd . "\n");
    }
}

main();
?>

File: LaboratoryWork/Semester_2/Laboratory_7/Laboratory_2/Rust/task3.rs
use std::io;

fn is_number(a: i32) -> bool {
    /* Проверяет, что это число, а не цифра. */
    a / 10 != 0
}

fn cnt_odd_digits(mut a: i32) -> i32 {
    /* Считает количество нечетных цифр в числе. */
    let mut res = 0;
    a = a.abs();

    while a > 0 {
        res += a % 2;
        a /= 10;
    }

    res
}

fn main() {
    let mut input = String::new();
    println!("Введите количество чисел: ");
    io::stdin().read_line(&mut input).expect("Ошибка ввода");

    let n: i32 = input.trim().parse().expect("Пожалуйста, введите число");

    for i in 0..n {
        let mut input = String::new();
        println!("Введите {} число: ", i + 1);
        io::stdin().read_line(&mut input).expect("Ошибка ввода");

        let num: i32 = input.trim().parse().expect("Пожалуйста, введите число");

        let mut cnt_odd = 0;
        if is_number(num) {
            cnt_odd = cnt_odd_digits(num);
        } else {
            println!("{} - это цифра!", num);
        }

        println!("Количество нечетных цифр: {}", cnt_odd);
    }
}
